embulk-standards/src/main/java/org/embulk/standards/JsonParserPlugin.java

```diff
@@ -129,18 +134,23 @@ public void run(TaskSource taskSource, Schema schema, FileInput input, PageOutpu
                     Value value;
                     while ((value = stream.next()) != null) {
                         try {
-                            if (!value.isMapValue()) {
-                                throw new JsonRecordValidateException(
-                                        String.format("A Json record must not represent map value but it's %s", value.getValueType().name()));
+                            final Iterable<Value> recordValues;
+                            if (task.getFlattenJsonArray()) {
+                                if (!value.isArrayValue()) {
```

Comment:
I a little bit thought about other types of "flattening", but it must be too much at the moment. It may be a bit worth just thinking an extensible configuration format, though.

(By other types of flattening, I mean, for example, doubly-nested arrays, or JSON objects with considering a key as a column.)

Reply:
As a result of f2f communication with @dmikurube, we will go with current version so far and consider such complex feature in the time when it's actually required.